card/lib/card/log.rb

```diff
@@ -266,18 +347,22 @@ def initialize( parent, level, args )
         @valid = true
         @parent = parent
         @children_cnt = 0
+        @children = []
         if @parent
-          @parent.add_children
+          @parent.add_children self
           #@sibling_nr = @parent.children_cnt
         end
       end
 
-      def add_children
+      def add_children child=false
         @children_cnt += 1
+        @children << child if child
       end
```

Comment:
For the text output the number of child used to be enough but to generate html output I need a proper tree structure.